icu4c/source/i18n/measunit_impl.h

```diff
@@ -18,6 +18,12 @@ U_NAMESPACE_BEGIN
 static const char16_t kDefaultCurrency[] = u"XXX";
 static const char kDefaultCurrency8[] = "XXX";
 
+struct U_I18N_API MeasureUnitImplWithIndex : public UMemory {
+    int32_t index;
+    LocalPointer<MeasureUnitImpl> unitImpl;
```

Comment:
Is there a benefit to making this a pointer rather than simply a value type?
A value type seems simpler to me, given no need to allocate or dereference the MeasureUnitImpl instance.

Reply:
Okay, MeasureunitImp is hard to copy and move (contains list of SingleUnits ... etc) Therefore, Using pointers is much efficient and I kinda like it :)

Reply:
I'm exploring this independently - if we were to add a normal copy constructor to MeasureUnitImpl (will have to discuss with Shane), then copy elision could make a value type quite efficient.

Reply:
In the meantime, the constructor needs a comment explaining ownership of the unitImpl parameter. I'm adding that.

Reply:
I added a comment in 99fdf1957e223b106eac4f7949f9bcd8780f5461.